ocaml/xapi/xapi_session.ml

```diff
@@ -725,6 +725,128 @@ let slave_local_login_with_password ~__context ~uname ~pwd =
   debug "Add session to local storage" ;
   Xapi_local_session.create ~__context ~pool:false
 
+module Caching = struct
+  type external_auth_result = {
+      subject: [`subject] Ref.t
+    ; subject_identifier: string
+    ; subject_name: string
+    ; rbac_permissions: string list
+  }
+
+  module type EXTERNAL_AUTH_CACHE =
+    Helpers.AuthenticationCache.S
+      with type user = string
+       and type password = string
+       and type session = external_auth_result
+
+  let create_salt =
+    let module MRNG = Mirage_crypto_rng in
+    let g = MRNG.Fortuna.create () in
+    let initial_seed =
+      let r = Random.int64 Int64.max_int in
+      let bs = Bytes.create 8 in
+      Bytes.set_int64_le bs 0 r ; Cstruct.of_bytes bs
+    in
+    MRNG.Fortuna.reseed ~g initial_seed ;
```

Comment:
Does this mean the `reseed` will be performed only once when the xapi is starting up?

Reply:
Indeed. From the https://github.com/mirage/mirage-crypto/blob/v0.11.3/rng/fortuna.ml#L97, it seems more entropy will be added if necessary in `add_pool_entropy`, which is called in `generate` always.